### PR TITLE
Rename svZeroDPlus to svZeroDSolver 

### LINKED ISSUE #35: Move svZeroDPlus to Simvascular and archive Simvascular/svZeroDSolver
### **This issue is a follow-up to #28 and the final step in making svZeroDPlus official.** 

_Below is a copy of the initial code and docs review by @ktbolt, copied over from #28._ 

 I had a look at some of the code and documentation, comments below.

**Documentation Review**

The `README` documentation needs to be improved 
- stating what is created when building: library and two executables
- a section describing the 3D solver interface

I can do this.

The `C++ Documentation` https://stanfordcbcl.github.io/svZeroDPlus/cpp/ needs to be improved 
- discusses Python
- can be organized a bit better

I can do this.

The documentation describing governing equations, code functionality, JSON file format, Python API, etc. needs to be in the SimVascular web site [here](https://simvascular.github.io/documentation/rom_simulation.html#0d-solver) not in github.io.

`Developer Guide` documentation is just a short _Architecture_ paragraph. There needs to be a _Design_ section that describes the problem, its components (i.e., objects) and how it is solved. Define any terms that will be used to relate the classes created in the implementation. 

There needs to be an _Implementation_ section that describes how the concepts in the `Design` section are realized in code. Describe the system structure (classes) and behavior, showing how the primary classes interact. Have a section describing how a developer can add new functionality.

`Code Comments`  
- some code (bloodvesseljunction.hpp) document equations which are not unreadable and essentially meaningless when a developer is looking in the source file. This kind of thing should be in the documentation. 
- don't need to add comments that are totally obvious (@brief Destroy the Block object)
- have comments of class method declaration rather than implementation

**Code Review**
I have identified a lot of issues in the code that need to be addressed. 

The primary issue (so far) in the implementation is the use of templates to define class float or double precision. There is no strong use-case for this and it makes the code cluttered and more complicated. So the first change to make is to remove all of the class templates and just use double.

I will list the other issues here after I finish my review of the code.

_Originally posted by @ktbolt in https://github.com/StanfordCBCL/svZeroDPlus/issues/28#issuecomment-1703950030_

### LINKED ISSUE #86: Change paths in documentation
We will have to replace a couple of mentions of `StanfordCBCL` with `SimVascular` throughout `docs` and `README.md` (#35). We should do this after the move so we can check if the links still work. I think our `CI` pipeline should be fine.

### PR DESCRIPTION
**Pull request by menon-karthik**
## Current situation
Addresses #86 and #35. The move from StanfordCBCL to SimVascular involved renaming the repo to svZeroDSolver for consistency. The name therefore had to be updated in several places within the repo. 

## Release Notes 
- Main change is the name being updated in the docs, cmake files, github actions, etc.
- Another important change is the renaming of the python interface. It used to be called `svzerodplus`. Since the name `svzerodsolver` is already taken by the main executable, I thought it would be a good idea to rename the python interface `pysvzerod`. Please let me know what you think @richterjakob!

## Documentation
 
This PR mostly updates the docs. No code changes. 

## Testing
Github actions are passing.

## Code of Conduct & Contributing Guidelines 
- I agree to follow the [Code of Conduct](https://github.com/SimVascular/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/SimVascular/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by menon-karthik** [comment]
@richterjakob I requested your review because I might have broken some things in the python interface and docker container while renaming things. I'm not sure if I did, because the tests are passing but I'm assuming the testing doesn't include the docker container. Please don't feel like you need to spend too much time on this, but just wanted to make sure those two things are working!

File: docs/pages/main.md [hunk 1]
```diff
@@ -44,13 +44,13 @@ You can find more details about governing equations in individual blocks, for ex
 - BloodVesselJunction
 - WindkesselBC
 
-For implementation details, have a look at the [source code](https://github.com/StanfordCBCL/svZeroDPlus).
+For implementation details, have a look at the [source code](https://github.com/StanfordCBCL/svZeroDSolver).
```
Comment (mrp089) [inline docs/pages/main.md]: StanfordCBCL -> SimVascular

File: docs/pages/main.md [hunk 2]
```diff
@@ -59,12 +59,12 @@ For a pip installation, simply run the following command
 (cloning of the repository is not required):
 
 ```bash
-pip install git+https://github.com/StanfordCBCL/svZeroDPlus.git
+pip install git+https://github.com/StanfordCBCL/svZeroDSolver.git
```
Comment (mrp089) [inline docs/pages/main.md]: StanfordCBCL -> SimVascular

**Review by mrp089 — CHANGES_REQUESTED** [review]
Thanks for making it happen, @menon-karthik! Let's have a release party in the new year!

Good from my side, just two more places to change paths. Maybe run one final `grep -ir stanfordcbcl` on the repository.

**Review by mrp089 — APPROVED** [review]

**Comment by menon-karthik** [comment]
@mrp089 Thanks for approving! I'll wait for @richterjakob to look at it whenever he gets a chance before merging (no hurry).

**Comment by mrp089** [comment]
Merging, as @richterjakob and I want to use the new name in https://github.com/StanfordCBCL/svSuperEstimator. He accepts the name changes with a tear in his eye.

**Pull request merged**